gdraw/gdraw.c

```diff
@@ -477,14 +476,40 @@ void GDrawScroll(GWindow w, GRect *rect, int32 hor, int32 vert) {
 
 /* draws the subset of the image specified by src starting at loc (x,y) */
 void GDrawDrawImage(GWindow w, GImage *img, GRect *src, int32 x, int32 y) {
-    GRect r;
-    if ( src==NULL ) {
-	struct _GImage *base = img->list_len==0?img->u.image:img->u.images[0];
-	r.x = r.y = 0;
-	r.width = base->width; r.height = base->height;
-	src = &r;
+    int width = GImageGetWidth(img);
```

Comment:
All the changes here and in charview.c are to fix what gets displayed when at that zoom level.